### PR TITLE
OpenBLAS, OpenBLAS-devel: fix CMake build for PowerPC

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

This finally fixes the build on PowerPC.
See:
https://github.com/OpenMathLib/OpenBLAS/pull/4370
https://github.com/OpenMathLib/OpenBLAS/issues/4366
https://trac.macports.org/ticket/68868

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by catap** [comment]
@barracuda156 your patches were merged, shall you just update devel port to reduce number of patches?

**Comment by barracuda156** [comment]
@catap Yes, sure, will be done once back to a desktop.

**Comment by barracuda156** [comment]
@catap Done. Let’s wait until CI pass, then should be good to merge.

**Comment by catap** [comment]
@barracuda156 you must bump non devel subport.

**Comment by barracuda156** [comment]
@catap Why though? All changes are exclusive to PowerPC, where it would not have built.

Having said that, I need to test something. Back to draft for a bit.

**Comment by barracuda156** [comment]
@catap @michaelld @NicosPavlov This should be good now.

**Review by catap — APPROVED** [review]

**Comment by barracuda156** [comment]
@michaelld @pmetzger @reneeotten Could someone please help merging this? Fixes are just backports from the upstream.

**Pull request merged**
